cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_machineset.go

```diff
@@ -119,3 +122,31 @@ func newMachineSetScalableResource(controller *machineController, machineSet *Ma
 		minSize:    minSize,
 	}, nil
 }
+
+func (r machineSetScalableResource) Labels() map[string]string {
+	return cloudprovider.JoinStringMaps(r.machineSet.Labels, r.machineSet.Spec.Template.Labels, r.machineSet.Spec.Template.Spec.Labels)
+}
```

Comment:
The only ones propagated on node creation are `machineDeployment.Spec.Template.Spec.Labels` so no need to join.

https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/machine/v1beta1/machine_types.go#L72-L81

Reply:
so, it turns out that reducing this to just `r.machineSet.Spec.Template.Spec.Labels` causes our test to not work. i'm not sure which is more correct though, keep the label join here or fix the test.